### ISSUE TITLE
Darktable 4.6.0 crashes when using 'tag' filter in the collections module

### ISSUE THREAD

**Issue by sweeoon** [darktable-org/darktable#16287]
### Describe the bug

When I select the "Tag" filter in the Collections module, Darktable crashes even before I had a chance to enter any keywords.

### Steps to reproduce

1. Go to "collections" module.
2. Click on filter selection drop down menu and select "tag"
3. Darktable crashes.

### Expected behavior

Darktable should allow entry of keywords and show results.

### Where did you obtain darktable from?

The flathub repository within the linux mint repository.

### darktable version

4.6.0

### What OS are you using?

Linux

### What is the version of your OS?

Linux Mint 21.3

### Describe your system?

Linux kernel 5.15.0-94 generic, 64GB RAM, RTX 3060 Ti GPU.
AMD Ryzen 7 2700X CPU.
System root volume has 153GB free space. Swap volume is 65GB.

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

nVidia RTX 3060 Ti, 8GB VRAM, driver version 535.154.05

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

No similar issues with Darktable version 3.8.
A fresh install of Darktable 4.6.0 from flathub repository does not resolve issues. All previous configs deleted before fresh install.

I don't use lua scripts.

**Comment by ralfbrown** [comment]
Unable to reproduce with either 4.6.0 or master.

Does this happen with a new config directory (run as `darktable --configdir <dirname>` where the given directory does not yet exist) before you import images?  if so, please provide the `data.db` which will be in that directory after just starting and exiting darktable.  If you need to import images for the problem to occur, you almost certainly have a corrupted sidecar file and will need to isolate which one by importing your images in stages until you get the crash again.

**Comment by gi-man** [comment]
Attaching a log using -d common might also help understand the source of the crash.

**Comment by MStraeten** [comment]
an issue report clearly wins if it has a meaningful title …

**Comment by sweeoon** [comment]
The issue with the "tag" filter in collections happened while I was using an existing database that has been accumulating for years. I had no issues with the collections module filters with previous darktable versions until 4.6.0.

I thought it might be a corrupted file somewhere, so I installed darktable 3.8 and imported my entire library again, with all the sidecar files, too. No problems. Then I tried to locate a few photos using the tag filter in collections. Again no problems. So I don't think it's a corrupted sidecar file or photo.

I also did a fresh install of the flatpak version of darktable 4.6.0 to check. It couldn't import my entire library using the "recursive" option, so I had to import each folder manually. It got through about a dozen folders and then it crashed.

I'm not a technical user, just someone on a linux computer editing photos. When darktable crashes, does it generate a log file? Where can I find that for the flatpak install of darktable 4.6.0?

**Comment by gi-man** [comment]
The FAQ has the details for generating a log file from flatpak.

https://www.darktable.org/about/faq/

**Comment by ralfbrown** [comment]
Do you remember which folder you imported when it crashed?  There's probably a sidecar in that directory which 4.6 is having problems with (even though it works fine for 3.8), so you should import that directory in stages to isolate the image(s) causing problems.  Ideally, you'll be able to find a single image which causes a crash when imported into a fresh configuration -- sharing the resulting `data.db` and any crash log will give devs the best chance of figuring out what is going wrong.

**Comment by Yu-Misaka** [comment]
I experienced a similar situation: my darktable crashed when I tried to switch collection filter to "folder". I located the folder causing the crash and solved the problem by renaming that folder and change its path. Hope this can help in your case.

I don't understand why it happend or how it is solved, but I suspect it might have something to do with the folder's name (maybe its name/path is too long, or contain certain unsupported character/pattern).

My crash report and data file were lost, I will try to share them if the crash happends again.
